### PR TITLE
Fix remove account flow

### LINKED ISSUE #4535: Adjust remove account flow
### Expected behavior
1- On remove an account if the current account is same account, user needs to be redirect to account management and current account should be erased
2- Cancel button on remove account modal doesn't work

### Actual behavior
1- account will be remove but user still using the same account,
2- modal should be close

### Steps to reproduce
- Add an account
- Select that account
- Remove the same account you're in

### Which version(s) does this affect? (Environment, OS, etc...)
v3.0.0

### PR DESCRIPTION
**Pull request by oskarleonard**
### What was the problem?

This PR resolves #4535

### How was it solved?

- By removing current account from the redux store when deleting an account. 
- Merging two different UI for performing the same thing

### How was it tested?

Manually

### CODE REVIEW

File: src/modules/account/components/RemoveAccount/RemoveAccount.test.js [hunk 1]
```diff
@@ -19,12 +19,12 @@ jest.mock('@account/hooks', () => ({
   )),
 }));
 
-describe('Remove account', () => {
+describe('Remove account ', () => {
```
Comment (ikem-legend) [inline src/modules/account/components/RemoveAccount/RemoveAccount.test.js]: @oskarleonard Please remove empty space

File: src/modules/account/components/AccountRow/AccountRow.test.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ jest.mock('react-i18next');
 
 const props = {
   account: mockSavedAccounts[0],
-  showRemove: true,
+  onRemove: jest.fn(),
```
Comment (ikem-legend) [inline src/modules/account/components/AccountRow/AccountRow.test.js]: @oskarleonard Use a constant for the onRemove mock.
For example, `const mockOnRemove = jest.fn() and then reuse it here and while rendering the component below
```suggestion
  onRemove: mockOnRemove,
```

File: src/modules/account/components/AccountRow/AccountRow.test.js [hunk 3]
```diff
@@ -27,9 +27,9 @@ describe('Select Account Row', () => {
     expect(container.getByText(truncateAddress(mockSavedAccounts[0].metadata.address))).toBeTruthy();
   });
 
-  it('Should  list', async () => {
+  it('Should show delete button', async () => {
     const container = render(<AccountRow {...props} />);
-    container.rerender(<AccountRow {...props} showRemove />);
+    container.rerender(<AccountRow {...props} onRemove={jest.fn()} />);
```
Comment (ikem-legend) [inline src/modules/account/components/AccountRow/AccountRow.test.js]: @oskarleonard 
```suggestion
    container.rerender(<AccountRow {...props} onRemove={mockOnRemove} />);
```

**Review by ikem-legend — APPROVED** [review]
Looks good to me

**Review by soroushm — APPROVED** [review]
LGTM

**Pull request merged**